modules/browser/detect_lastpass/command.js

```diff
@@ -4,26 +4,41 @@
 // See the file 'doc/COPYING' for copying permission
 //
 
+$j = jQuery.noConflict();
+
 beef.execute(function() {
-	var result = "Not in use or not installed";
+	var result = "Module - Detect LastPass: LastPass Not in use or not installed";
 
-	var lpdiv = document.getElementById('hiddenlpsubmitdiv');
-	if (typeof(lpdiv) != 'undefined' && lpdiv != null) {
-		//We've got the first detection of LP
-		result = "Detected LastPass through presence of the <script> tag with id=hiddenlpsubmitdiv";
-	} else if ($j("script:contains(lastpass_iter)").length > 0) {
-		//We've got the second detection of LP
-		result = "Detected LastPass through presense of the embedded <script> which includes references to lastpass_iter";
-	} else {
+	// The following base64 encoded string represents the LastPass inline PNG which is inserted into user/pass form fields
+	var base64PNG = "iVBORw0KGgoAAAANSUhEUgAAABAAAAASCAYAAABSO15qAAAAAXNSR0IArs4c6QAAAPhJREFUOBHlU70KgzAQPlMhEvoQTg6OPoOjT+JWOnRqkUKHgqWP4OQbOPokTk6OTkVULNSLVc62oJmbIdzd95NcuGjX2/3YVI/Ts+t0WLE2ut5xsQ0O+90F6UxFjAI8qNcEGONia08e6MNONYwCS7EQAizLmtGUDEzTBNd1fxsYhjEBnHPQNG3KKTYV34F8ec/zwHEciOMYyrIE3/ehKAqIoggo9inGXKmFXwbyBkmSQJqmUNe15IRhCG3byphitm1/eUzDM4qR0TTNjEixGdAnSi3keS5vSk2UDKqqgizLqB4YzvassiKhGtZ/jDMtLOnHz7TE+yf8BaDZXA509yeBAAAAAElFTkSuQmCC";
+
+	while (result)
 
-		//Form is not there, lets check for any form elements in this page, because, LP won't activate at all without a <form>
-		if (document.getElementsByTagName("form").length == 0) {
-			//No forms
-			result = "The page doesn't seem to include any forms - we can't tell if LastPass is installed";
+	// Detect input form fields with the injected LastPass PNG as background image
+	var bginput =  $j('input[style]');
+	var lpdiv = document.getElementById('hiddenlpsubmitdiv');
+	if (bginput.length > 0) {
+		beef.debug("input field with 'style' attribute found: " + bginput);
+		for(var i = 0; i < bginput.length; i++) {
+			beef.debug('Module - Detect LastPass: Number of potential input fields with \'style\' attribute found: ' + bginput.length);
+			var styleContent = bginput[i].getAttribute('style');
+			if (styleContent.includes(base64PNG)) {
+				beef.debug('Module - Detect LastPass: Matching inline PNG detected');
+				result = "Module - Detect LastPass: Detected LastPass through presence of inline base64-encoded PNG within input form field";
+			}
 		}
-		
-	}
+	// Detect presence of LastPass iframe
+	} else if ($j("iframe[name='LPFrame']").length > 0) {
+		beef.debug('Module - Detect LastPass: Matching iframe found');
+		result = "Module - Detect LastPass: Detected LastPass through presence of LastPass 'save password' iframe";
+	// Below is the older method of LastPass detection method
+	} else if (typeof(lpdiv) != 'undefined' && lpdiv != null) { 
+		// We've got the first detection of LP
```

Comment:
This is no longer the first detection. And the comment is redundant - detection is implied by setting `result`.

```suggestion
```